waspc/test/JsImportTest.hs

```diff
@@ -0,0 +1,53 @@
+module JsImportTest where
+
+import StrongPath (Dir', File', Path, Posix, Rel, (</>))
+import qualified StrongPath as SP
+import Test.Tasty.Hspec (Spec, describe, it, shouldBe)
+import Wasp.JsImport
+
+spec_JsImportTest :: Spec
+spec_JsImportTest = do
+  describe "makeJsImport" $ do
+    it "makes JsImport with default import from a path" $ do
+      makeJsImport pathToJsFile (JsImportModule "test")
+        `shouldBe` JsImport
+          { _path = pathToJsFile,
+            _name = JsImportModule "test",
+            _importAlias = Nothing
+          }
+  describe "applyJsImportAlias" $ do
+    it "applies alias to JsImport" $ do
+      applyJsImportAlias
+        (Just "alias")
+        (makeJsImport pathToJsFile (JsImportModule "test"))
+        `shouldBe` JsImport
+          { _path = pathToJsFile,
+            _name = JsImportModule "test",
+            _importAlias = Just "alias"
+          }
```

Comment:
What about:

```suggestion
      let jsImport = makeJsImport pathToJsFile (JsImportModule "test")
      applyJsImportAlias (Just "testAlias") jsImport
        `shouldBe` jsImport { _importAlias = Just "testAlias" }
```

This way you are testing exactly what you wanted to test. Yes, this comes down to replicating implementation, but that is because implementation is trivial. When implementation is super trivial, one might wonder is test even useful / needed? The only thing that test does is just repeat implementation. This does make it more robust to unexpected changes -> if you specify in two places how something works, you need to change implementation in both places, which means it is resistant to mistake / accidental change in just one place. So I would argue it can be ok to do a test even for such trivial functions, although I would probably also be ok with even dropping the test.

Reply:
Btw, one way to make this test better is to actually use property testing (QuickCheck).
I think this is a good opportunity to go for it, give it a try. You would basically define Arbitrary class for JSImport and somewhat modify this test then to test for many different possible js imports.

Reply:
I'm disliking test results that have *any* logic or computation inside. That's why I've went with the verbose construction of `JsImport` and repeated result strings in many places in the `getImportStmtAndIdentifier` tests.

Reply:
We talked about this offline, and we agreed that using a setter it's okay to construct the expected data. The aim is to keep the tests maintaintable and making the code, cleaner.

Reply:
We agreed not to do property testing here since it's hard to define the properites to test and we already test all of the behaviours we need.